src/mame/jaleco/tetrisp2_v.cpp

```diff
@@ -156,7 +159,13 @@ TILE_GET_INFO_MEMBER(tetrisp2_state::get_tile_info_fg)
 
 void tetrisp2_state::tetrisp2_vram_fg_w(offs_t offset, u16 data, u16 mem_mask)
 {
-	COMBINE_DATA(&m_vram_fg[offset]);
+	// VJ and Stepping Stage write to the upper byte here to display ASCII text,
+	// other usages in those games outside of ASCII text write a full 16-bit value.
+	if (ACCESSING_BITS_0_7)
+		m_vram_fg[offset] = data;
+	else
+		m_vram_fg[offset] = data >> 8;
+
```

Comment:
Byte smearing? cfr. PR #10957

Reply:
Had to look that term up. Yes, that's my guess. 

This technique is also already used for the priority data in tetrisp2_v.cpp as well:
https://github.com/mamedev/mame/blob/master/src/mame/jaleco/tetrisp2_v.cpp#L86-L105

Reply:
You shouldn’t need that voodoo though – byte smearing is implemented correctly in the 68k CPU device.  You should be able to just ignore `mem_mask`.

Reply:
This isn't a write into an 68K CPU, it's a write into the sprite chip which I don't know what it uses on the main PCB (the ones on the subboards are implemented in FPGAs). In the issue I was trying to addresses case it's also a 8-bit write into the upper half of a 16-bit register so if you ignore `mem_mask` you'll end up with a value of 0xff00 instead of the actual value of 0xff. So the code is still broken.

![0004](https://user-images.githubusercontent.com/63495610/226457027-76cb423c-059a-425b-ad04-6d3f90c20ac1.png)

vs the expected

![0008](https://user-images.githubusercontent.com/63495610/226457557-a75fae08-99d8-4185-9035-76281436ba12.png)

Reply:
But isn’t it a write *from* the 68k CPU?  When a 68k does a byte write into the MSB, you'll get the smeared value in `data`.  For example writing 0xab to the MSB will give you `data` = 0xabab and `mem_mask` = 0xff00.  The NuBus MDC video card firmware depends on this when doing palette writes – it sometimes writes to the MSB and sometimes writes to the LSB, and depends on the card always taking what’s on the LSB and ignoring lane select.

Are you sure it isn’t just that the tilemap memory is only eight bits wide?  What happens if you just do `m_vram_fg[offset] = data & 0x00ff;`?

Reply:
In fact, I’m pretty sure that’s all it is – it’s just an 8-bit peripheral on the 16-bit 68k bus and it ignores lane select altogether.

Reply:
https://github.com/mamedev/mame/blob/52e150bcfd4128784bb73fb418ed076c99e887c1/src/mame/jaleco/ms32.cpp#L633

Been known to do this stunt on the original HW as well (tetrisp2.cpp is a spin-off)

Reply:
`m_vram_fg[offset] = data & 0x00ff;` won't work either. On the operator menu everything uses the default palette so it doesn't show up there, but if you go into actual gameplay it *does* use the full 16-bits to include palette data so you get this mess. Other than that, the operator menu uses 8-bit writes (mask is ff00) *except* for the numbers on the sound check menu which are full 16-bit writes.

![0012](https://user-images.githubusercontent.com/63495610/226463771-df9cecb1-7d3a-40d5-bf08-4d78b677988b.png)

vs the expected

![0010](https://user-images.githubusercontent.com/63495610/226463801-9e12ba40-cad0-4dad-8a4e-2c1a48a2e575.png)

Reply:
I re-added a check on the lane select lines.  Sorry, been multi-tasking and not checking all the games.

Reply:
Latest commit looks fine to me in all of the test cases I know of. 👍

Reply:
interesting is that Jaleco and NMK were often very close in terms of hardware, makes me wonder if those NMK games could be a more complex case too if it's some common design pattern.